packages/cw721/src/execute.rs

```diff
@@ -0,0 +1,810 @@
+            MINTER.initialize_owner(storage, api, Some(legacy_minter.as_str()))?;
+            CREATOR.initialize_owner(storage, api, Some(legacy_minter.as_str()))?;
+            Some(legacy_minter.to_string())
+        }
+    };
+    Ok(response.add_attribute("creator_and_minter", none_or(creator_and_minter.as_ref())))
+}
+
+/// Migrates only in case collection_info is not present
+pub fn migrate_legacy_collection_info(
+    storage: &mut dyn Storage,
+    env: &Env,
+    _msg: &Cw721MigrateMsg,
+    response: Response,
+) -> Result<Response, Cw721ContractError> {
+    let contract = Cw721Config::<
+        DefaultOptionMetadataExtension,
+        Empty,
+        Empty,
+        Empty,
+        DefaultOptionCollectionInfoExtension,
+    >::default();
+    match contract.collection_info.may_load(storage)? {
+        Some(_) => Ok(response),
+        None => {
+            // contract info is legacy collection info
+            let legacy_collection_info_store: Item<cw721_016::ContractInfoResponse> =
+                Item::new("nft_info");
+            let legacy_collection_info = legacy_collection_info_store.load(storage)?;
+            let collection_info = CollectionInfo {
+                name: legacy_collection_info.name.clone(),
+                symbol: legacy_collection_info.symbol.clone(),
+                extension: None,
+                updated_at: env.block.time,
+            };
+            contract.collection_info.save(storage, &collection_info)?;
+            Ok(response
+                .add_attribute("migrated collection name", legacy_collection_info.name)
+                .add_attribute("migrated collection symbol", legacy_collection_info.symbol))
+        }
+    }
+}
+
+/// Migrates only in case collection_info is not present
+pub fn migrate_legacy_tokens(
+    storage: &mut dyn Storage,
+    _env: &Env,
+    _msg: &Cw721MigrateMsg,
+    response: Response,
+) -> StdResult<Response> {
+    let contract =
+        Cw721Config::<DefaultOptionMetadataExtension, Empty, Empty, Empty, Empty>::default();
+    match contract.nft_info.is_empty(storage) {
+        false => Ok(response),
+        true => {
+            let indexes = TokenIndexes {
+                owner: MultiIndex::new(token_owner_idx, "tokens", "tokens__owner"),
+            };
+            let legacy_tokens_store: IndexedMap<
+                &str,
+                NftInfo<DefaultOptionMetadataExtension>,
+                TokenIndexes<DefaultOptionMetadataExtension>,
+            > = IndexedMap::new("tokens", indexes);
+            let keys = legacy_tokens_store
+                .keys(storage, None, None, Order::Ascending)
+                .collect::<StdResult<Vec<String>>>()?;
+            for key in keys {
+                let legacy_token = legacy_tokens_store.load(storage, &key)?;
+                contract.nft_info.save(storage, &key, &legacy_token)?;
+            }
+            Ok(response)
+        }
+    }
+}
```

Comment:
This is prob not needed, it does so since key `tokens` is renamed to `nft`. So maybe remove this. Unless we also want to add `updated_at` to NftInfo - if needed.

Reply:
Removed - no need to rename token keys. It takes too long for migration and impacts gas fees.